CODE_REVIEW.md

```diff
@@ -0,0 +1,65 @@
+# Code reviews and PR process
+
+We use PR reviews to achieve the following purposes:
+  1. Quality: Increase the quality of the software and, as a consequence, of our product
+  2. Training/Coaching/Mentoring: we improve reviewed and reviewers knowledge and skills
+  3. Knowledge sharing: we align the team about technical backgrounds and information
+  4. Idea gatering: we discuss, share and evaluate ideas that creates opportunities
+
+As a developer you are responsible for the quality of your PR and for the amount of time the PR stays open.
+Your PR needs to pass all the reviews steps as fast as possible, you have to communicate and urge the reviewers everytime the PR needs to be reviewed.
+You are invited to read carefully your PR before request a review, this can drastically reduce the time your PR stays open before being approved.
+Reading multiple times your PR changes in diff tools like github can help you spot parts that can be improved.
+
+## PR steps
+### Open a PR
+
+Be sure you are providing the necessary information to speed-up the PR process an to help reviewers doing their part:
+
+* Title should contain the reference to the story/ticket if existing
+* Title should summarise the purpose of the implementation by the code POV, not replicate the title of the story that represent usually the user POV
+* Description
+  1. Should contain link to useful references or resources if available (Jira ticket or tickets, libraries, external documentation...)
+  2. You should provide the right context and background information to help reviewers undestand your implemntation choices, avoided un-necessary subsequent questions ('I used xxx because of xyz...')
+  3. Tests plan: You should provide a basic tests plan. This is to communicate to reviewers that you knew how the feature is expected to work and can also help human QA.
+
+### Assign reviewers
+
+PRs need to be **'peer approved'** and **'core approved'**.
```

Comment:
Not all teams are big so I would rather make this a team way thing. To make it generic we basically put it based on team size. 

Example : On Mobile the GBR team is 4 guys and 1 for every platform. Though we do cross review its never enough. I include the other member of the different team to do reviews but it can be difficult. 

So it depends on team size.

Reply:
Also this is more team related if here we are trying to set the process which is a skeleton in my opinion ?

Reply:
I feel creating PR adding labels comments etc. 

We put in examples based on certain parameters which we can add ?

Reply:
I guess this is a best practice applicable to a team of 3+ non-functional developers.
I think that also if the devs are not-functionally related (they don't have the same skills set) it's important to have at least 2 reviewers. In case of non-functional developers, the review process can be anyway useful to align the team in term of 'big picture' and contaminate the reviewers with something coming from other technologies.
Let's say your team is you, an adroid dev and an ios dev. IMO you shoud anyway have 2 reviewers cause the devs can learn from each other and you, as a core reviewer can use the PR to communicate with both about business logic choices that need to be shared across the whole team.
Does this make sense?

Reply:
If you agree I can review my PR with this concepts. I think they are important.

Reply:
The labelling system doesn't need to be strict but the most we are aligned across team the simpler is the turnover.
I think we should think it as a basic set in which every team can build a superset of labels.

Reply:
Agreed. Lets put this as a Good to Have to keep the flat structure.

Reply:
Late to the convo, but agree two reviews don't make sense on smaller/newer teams and don't always make sense for some PRs. We do a mix on the CP, with details as to when to use both, as you've probably seen here: https://docs.google.com/document/d/1_yYFSeazyjMYjwp7xkZVj_IIAPRyYe4oDTjsqMcyg7o/edit?usp=sharing

So agree that we put the two reviews as a recommended when possible and outline the difference in the two and when it is best to use two reviews.

Reply:
Ok, will try to adapt it with your suggestions.